### PR TITLE
Added projection test cases for $slice

### PR DESCRIPTION
**Pull request by vic-tsang**
This PR contains: 
- projection test cases for $slice
- introduced skip_rejection_types for BSON comparison test cases as the engine converted some of the projection slice into valid expression slice which didn't result in error (the skipped test cases are tested outside the framework from line 189 in test_slice_projection_behavior.py)
- renamed error code EXPRESSION_SLICE_NOT_ARRAY_ERROR to SLICE_INVALID_ARGUMENT_ERROR

Ref: 
- Issue #29

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_projection_behavior.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
+"""Tests for $slice projection: inclusion/exclusion, nested paths, and edge cases.
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_projection_behavior.py:1]: MongoDB docs state $slice projection works in both find and findAndModify. All tests use find only. Consider adding one findAndModify test to confirm $slice projection works there (e.g., findAndModify with fields: {"arr": {"$slice": 1}}).

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    ProjectionTestCase(
+        "fractional_limit_truncates_to_zero_error",
+        projection={"arr": {"$slice": [0, 0.5]}},
+        doc=[{"_id": 1, "arr": [1, 2, 3]}],
+        error_code=SLICE_INVALID_ARGUMENT_ERROR,
+        msg="$slice: [0, 0.5] should error — fractional limit truncates to 0 which is invalid",
+    ),
+]
+
+SLICE_RESTRICTION_ERROR_TESTS: list[ProjectionTestCase] = [
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py:85]: The docs call out `{"instock.$": {"$slice": 1}}` as invalid — this is `$slice` used as the value of a `$` positional projection key. Your existing test `{"arr": {"$slice": 1}, "arr.$": 1}` tests them as two separate sibling projection entries (path collision). The docs example is a different syntax: `$slice` nested inside the positional expression. Consider adding `{"arr.$": {"$slice": 1}}` with a matching filter to cover this.

**Review by fanyangv — CHANGES_REQUESTED** [review]

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_projection_behavior.py (hunk 1 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_projection_behavior.py:1]: added findAndModify tests

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py (hunk 2 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py:85]: added test_slice_inside_positional_projection_error to test spefically for "projection": {"arr.$": {"$slice": 1}}

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    )
+    assertFailureCode(result, test.error_code, msg=test.msg)  # type: ignore[arg-type]
+
+
+VIEW_ERROR_TESTS: list[ProjectionTestCase] = [
+    ProjectionTestCase(
+        "view_restriction",
+        projection={"arr": {"$slice": 1}},
+        doc=[{"_id": 1, "arr": [1, 2, 3]}],
+        expected=[{"_id": 1, "arr": [1]}],
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py]: expected not used.

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_single_value.py [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        projection={"arr": {"$slice": -1}},
+        doc=[{"_id": 1, "arr": []}],
+        expected=[{"_id": 1, "arr": []}],
+        msg="$slice: -1 on empty array should return empty array",
+    ),
+]
+
+NON_ARRAY_FIELD_TESTS: list[ProjectionTestCase] = [
+    ProjectionTestCase(
+        "field_is_null",
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_single_value.py:134]: 🟡 Minor — duplicate cases in NON_ARRAY_FIELD_TESTS (test_slice_single_value.py:132-189)
  - field_is_null (line 134) and null_vs_missing_null (line 176) assert the same thing — $slice on a null
  field returns null. The only difference is an unrelated extra field in the second doc.
  - field_missing (line 141) and null_vs_missing_absent (line 183) likewise duplicate — both assert that
  an absent field stays omitted.

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py (hunk 3 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_errors.py]: removed

File: documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_single_value.py (hunk 4 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/projection/slice/test_slice_single_value.py:134]: removed

**Review by eerxuan — APPROVED** [review]
